lib/src/editor.dart

```diff
@@ -0,0 +1,788 @@
+SourceEdit _removeFromFlowMap(
+    String yaml, YamlMap map, YamlNode keyNode, YamlNode valueNode) {
+  final keySpan = keyNode.span;
+  final valueSpan = valueNode.span;
+  var start = keySpan.start.offset;
+  var end = valueSpan.end.offset;
+
+  if (deepEquals(keyNode, map.nodes.keys.first)) {
+    start = yaml.lastIndexOf('{', start) + 1;
+    end = yaml.indexOf(RegExp(r',|}'), end) + 1;
+  } else {
+    start = yaml.lastIndexOf(',', start);
+  }
+
+  return SourceEdit(start, end - start, '');
+}
+
+/// Performs the string operation on [yaml] to achieve the effect of removing
+/// the [key] from the map, bearing in mind that this is a block map.
+SourceEdit _removeFromBlockMap(
+    String yaml, YamlMap map, YamlNode keyNode, YamlNode valueNode) {
+  var keySpan = keyNode.span;
+  var valueSpan = valueNode.span;
+  var start = yaml.lastIndexOf('\n', keySpan.start.offset);
+  var end = yaml.indexOf('\n', valueSpan.end.offset);
+
+  if (start == -1) start = 0;
+  if (end == -1) end = yaml.length - 1;
+
+  return SourceEdit(start, end - start, '');
+}
+
+/// Returns a safe string by checking for strings that begin with > or |
+String _getSafeString(String string) {
+  if (string.startsWith('>') || string.startsWith('|')) {
+    return '\'$string\'';
+  }
+
+  return string;
+}
+
+/// Returns values as strings representing flow objects.
+String _getFlowString(Object value) {
+  return _getSafeString(value.toString());
+}
+
+/// Returns values as strings representing block objects.
+// We do a join('\n') rather than having it in the mapping to avoid
+// adding additional spaces when updating rather than adding elements.
+String _getBlockString(Object value,
+    [int indentation = 0, int additionalIndentation = 2]) {
+  if (value is List) {
+    return value.map((e) => ' ' * indentation + '- $e').join('\n');
+  } else if (value is Map) {
+    return value.entries.map((entry) {
+      var result = ' ' * indentation + '${entry.key}:';
+
+      if (!_isCollection(entry.value)) return result + ' ${entry.value}';
+
+      return '$result\n' +
+          _getBlockString(entry.value, indentation + additionalIndentation);
+    }).join('\n');
+  }
+
+  return _getSafeString(value.toString());
+}
+
+/// Returns the content sensitive ending offset of a node (i.e. where the last
+/// meaningful content happens)
+int _getContentSensitiveEnd(YamlNode yamlNode) {
+  if (yamlNode is YamlList) {
+    return _getContentSensitiveEnd(yamlNode.nodes.last);
+  } else if (yamlNode is YamlMap) {
+    return _getContentSensitiveEnd(yamlNode.nodes.values.last);
+  }
+
+  return yamlNode.span.end.offset;
+}
+
+/// Checks if the item is a Map or a List
+bool _isCollection(Object item) => item is Map || item is List;
+
+/// Wraps [value] into a [YamlNode].
+YamlNode _yamlNodeFrom(Object value) {
+  if (value is Map) {
+    return YamlMap.wrap(value);
+  } else if (value is List) {
+    return YamlList.wrap(value);
+  } else {
+    return YamlScalar.wrap(value);
+  }
+}
```

Comment:
What if `value is YamlNode` then we probably don't need a transformation...

I don't know if this the case everywhere..

Reply:
Added an extra if statement to handle it!